ledger/acctupdates.go

```diff
@@ -316,6 +316,9 @@ func (au *accountUpdates) loadFromDisk(l ledgerForTracker, lastBalancesRound bas
 
 // close closes the accountUpdates, waiting for all the child go-routine to complete
 func (au *accountUpdates) close() {
+	au.accountsMu.Lock()
+	defer au.accountsMu.Unlock()
```

Comment:
If we have a race condition here, then we must have a bigger problem. 
I think it is the test which is wrong and the fix should be there. 

We do not need to put a lock here. `close()` is expected to only be called after no go-routine is using it and no go-routine will access it in the future.

The `au.accountsq` below is set to `nil`. By solving the race condition here, you might bump into accessing a `nil` `au.accountsq`... and panic !

Reply:
this is a very good observation. syncer belongs to trackers registry that first waits syncer to exit and then closes tracker, but this test (and some other tracker tests) close trackers directly. I'll fix that.